source/Gamerules.h

```diff
@@ -0,0 +1,42 @@
+/* Gamerules.h
+Copyright (c) 2022 by Amazinite
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef GAMERULES_H_
+#define GAMERULES_H_
+
+class DataNode;
+
+
+// Gamerules contains a list of constants and booleans that define game behavior,
+// for example, the spawnrate of person ships or whether universal ramscoops are active.
+class Gamerules {
+public:
+	Gamerules() = default;
+
+	// Load a gamerules node.
+	void Load(const DataNode &node);
+
+	int PersonSpawnrate() const;
+	bool UniversalRamscoopActive() const;
+
+
+private:
+	bool universalRamscoop = true;
```

Comment:
```suggestion
	bool universalRamscoop = false;
```

Universal ramscoop is a lore issue, not just a convenience/safety factor. Giving our players the option to have it enabled is good; but the default state should be without it.

Reply:
This is something that will be trivial to change by anyone of the community when we do add some mechanics to compensate. The last spaceport save is something in the right direction, but not sufficient.
Also since an issue was open instantly to change that, this obviously isn't a change everyone would agree to.

The best example of "lore" change that ruined the game experience would be the removal of portraits.

Reply:
We have *four* backup saves to prevent getting locked out, and better safety mechanisms have been discussed and are hopefully being worked on that make for *better* gameplay.

I'll note that in EV/O/N running out of fuel after having landed on a *single* uninhabited planet was a hard lock that led to player's having to either edit their save (not very easy to do) or start a new pilot, and yet it remained a staple of the franchise through two additional sequels. Given the volumes of people who successfully played the trilogy and loved it, I really don't think our new players need this hand holding. Especially since pretty much everywhere a new player can reach in the first few hours (baring considerable luck) is going to have *plenty* of traffic to help get them unstuck in an immersive, in-character, way. New players are no less capable of us, and depriving them of the opportunity to overcome this is worse than it'd be to remove pirates from the early game. Worse, because fuel is a challenge that a new player *can* overcome, whereas pirates are something they'll usually have to just run from for quite a while. Having meaningful challenges that they can overcome *is* an important part of gameplay. Yes, it is an easy one, as befits something that should mainly be a challenge for the very early game. (there are, after all, *lots* of ways to mitigate or even eliminate this risk, some of which are available from day 1.

Reply:
To be clear I'm not saying no never, I'm saying first have more spawnratea when low on fuel or anything else really to compensate

Reply:
The community doesn't have a *concensus* on the change, agreed. However, I think it's a change that we should keep and people should actually try out, rather than just blindly revert.

I'm quite serious when I say that I feel this makes the game *better.* As in all-round better gameplay, better experience, better story, better challenge, better respect for the player.

There's a reason why not only it was the case in EV, it was *kept* through two sequels, even though both EVO and EVN had *far* more uninhabited systems that one could get stuck in. A lot more long stretches that required ships to actually make an *effort* to have above average fuel capacity, too. They had the programming ability to change it if people running out of fuel had been a problem.

Reply:
The point of gamerules is to customize one's own experience. If *you* think that this gamerule is better set to false then *you* should customize your experience to do so. But this behavior has been true since v0.9.0, so that, plus the non-insignificant number of people who commented after #7730 was merged, suggests to me that with the introduction of the gamerule, its state should default to true.

Remember too that this PR is only a foundation. While at the moment it won't be easy for everyone to change this gamerule, it being a gamerule at least makes it several times more customizable than before, and it will become easier for people to mess with in time as gamerules start to be able to be customized in more places.

I also think it's important to note, because you do this a lot, that while Endless Sky is inspired by Escape Velocity, it is not itself Escape Velocity. Sure, we can look to EV for how it did things, but I don't think we should say "Because EV did this, we should do this." Now you might not be saying that exactly, but you do often use the fact that EV did things a certain way as part of an argument for why we should do things a certain way. We are a different game with different systems. Some similar, some not so similar, but at the end of the day we are a different game. Just because EV did something doesn't mean it's better and therefore should be done here. Any changes made to ES need to be justified *within* ES.

Reply:
Also, getting trapped on Vellos sucked. Getting trapped on Kontik sucked. Even if EV Nova had multiple previous saves, those are unnecessary gotchas for new players. Having the emergency ramscoop doesn't mean players won't use ramscoops later on, especially with out deep in finding the Ember Wastes is for a normal playthrough.

I'm kinda shocked to find out that removing the ramscoop was strong-armed like this. Default it on. I still had a wonderful time exploring the Ember Wastes the first time. Having a very minor safety net did not impact my enjoyment of it at all.

Reply:
I reference EV/O/N because it is over two decades of combined gameplay that we can learn from. We're already significantly better than it in many ways, but fuel management isn't one of them. Creating methods to prevent lock-out? Great! Awesome! I'm all in favor of that. Giving all ships infinite fuel is a *terrible* way to do that.

And we've already fixed that so players *can't* get stuck out there. Vellos and Kontik player-death literally can't happen in ES now. There's no need to hand-hold them the entire way.

And once again, I'll just point out: I'm a content creator writing lore for the story. I have to take into account the typical environment people play in while writing that story. Fuel scarcity is, actually, going to be an upcoming story arc. Or at least, that's what I have roughed out. Universal ramscoop makes a mockery of that. I'm not going to create a situation where the player can go "Hey, no problem, I'll just take off, make you 20-30 jumps of fuel, and here you go." Hey, if I can do that, why can't you? Hey, I must be special or something!"

Reply:
Beyond that, whatever happened to "Let's just let it sit and see what people think after experiencing it?" This is, after all, the continuous build. It's not even scheduled to be pushed out to an *unstable* release untill... January, maybe? Possibly February? There's a *lot* of time to have a month or two to let people actually try it out, implement various extra safety things, etc.

And before you say "they can try it out as a gamerule." I'll just state that applies just as much to trying out PRs, and we all know that doesn't happen. As Amazinite pointed out on Discord (roughly paraphrasing): "No matter how long a PR sits, the most likely time for negative reactions is immediately after it is merged, because *that* is when all the eyes actually look at how things changed." So let's *not* just leap to roll back the change into an invisible setting that the majority of people will never know is there and never change, and see how people actually handle it. See if an of the people on the cutting edge of continuous *actually* get stuck. I know I haven't so far (despite running out of fuel in uninhabited systems about a dozen times now, not a single reload because of it.).

Reply:
From the lore perspective, ramscoop outfits aren't exactly scarce - a 10 ton outfit available in human space from the start of the game for 60,000 credits might be outside the scope for a new player who is working off their loan (which is an important demographic, possibly the most important, to consider with regards to universal ramscoop) or a player still flying a smaller ship without 10 tons to spare, but it's not something that governments, factions or even well-off individual citizens would struggle to get their hands on.
And that's before considering that quite a few ships have built-in ramscoops anyway. So I don't understand how universal ramscoop (which is still reduced to 0 in any system where the effect of a ramscoop outfit would be reduced to 0 - it doesn't ignore changes to solar wind) could break a planned arc around fuel scarcity without ramscoop outfits or the built-in ramscoop of, for example, every korath automata warship also breaking it.

Reply:
I feel like we want to just get gamerule code in first so other PRs could utilize it. The universal ramscoop is clearly too controversial atm and it will just unnecessarily block this useful code feature. Probably best to just keep it to what's already in the game and discuss about removing or keeping it somewhere else.

Reply:
Is it controversial though? I feel like only Zitchas wants it off
And since realism is not a valid reason it should be on.
 If you want to remove it permanently go open an issue about it like smbd did to bring it back. I dont expect many will be for it, it has already worsened my game experience for sure.
And it seems like what Zitchas wants is just to disable it for the ember wastes not for the whole game. I could easily code that later on.

Reply:
It sparked entire poll to be made about merging process as well as enough talk here for Github to collapse some of the conversation. Personally I don't care too much about it except now it's blocking a generally useful PR. I just want to see this PR in first so other gamerule stuffs can follow and people who want to fight about the ramscoop thing can go do it somewhere else. 
Overriding gamerule per system sounds interesting though, but also sounds a little odd when its a universal rule and somehow these systems just say no but that's basically Ember Waste anyway so it's cool I guess.

Reply:
> Personally I don't care too much about it except now it's blocking a generally useful PR.
I don't think that's what's blocking this PR. It just needs some more reviews and approvals before merging.